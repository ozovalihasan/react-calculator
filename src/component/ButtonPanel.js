import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = ({ clickHandler }) => (
  <>
    <div className="button-panel main">
      <Button name="AC" clickHandler={() => clickHandler('AC')} />
      <Button name="+/-" clickHandler={() => clickHandler('+/-')} />
      <Button name="%" clickHandler={() => clickHandler('%')} />
      <Button name="รท" color="orange" clickHandler={() => clickHandler('รท')} />
    </div>
    <div className="button-panel main">
      <Button name="7" clickHandler={() => clickHandler('7')} />
      <Button name="8" clickHandler={() => clickHandler('8')} />
      <Button name="9" clickHandler={() => clickHandler('9')} />
      <Button name="X" color="orange" clickHandler={() => clickHandler('X')} />
    </div>
    <div className="button-panel main">
      <Button name="4" clickHandler={() => clickHandler('4')} />
      <Button name="5" clickHandler={() => clickHandler('5')} />
      <Button name="6" clickHandler={() => clickHandler('6')} />
      <Button name="-" color="orange" clickHandler={() => clickHandler('-')} />
    </div>
    <div className="button-panel main">
      <Button name="1" clickHandler={() => clickHandler('1')} />
      <Button name="2" clickHandler={() => clickHandler('2')} />
      <Button name="3" clickHandler={() => clickHandler('3')} />
      <Button name="+" color="orange" clickHandler={() => clickHandler('+')} />
    </div>
    <div className="button-panel main">
      <Button name="0" wide clickHandler={() => clickHandler('0')} />
      <Button name="." clickHandler={() => clickHandler('.')} />
      <Button name="=" color="orange" clickHandler={() => clickHandler('=')} />
    </div>
  </>
);

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
