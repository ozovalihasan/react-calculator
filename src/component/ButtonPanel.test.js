import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import ButtonPanel from './ButtonPanel';

let container = null;
let clickHandler;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  clickHandler = jest.fn();
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('shows different 19 buttons', () => {
  act(() => {
    render(<ButtonPanel clickHandler={clickHandler} />, container);
  });
  expect(container.querySelectorAll('button').length).toEqual(19);
  expect(container.textContent).toBe('AC+/-%รท789X456-123+0.=');
});

it('renders correctly', () => {
  const tree = renderer
    .create(<ButtonPanel clickHandler={clickHandler} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
