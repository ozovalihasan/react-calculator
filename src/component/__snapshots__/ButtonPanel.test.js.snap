// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="button-panel main"
  >
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      รท
    </button>
  </div>,
  <div
    className="button-panel main"
  >
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      X
    </button>
  </div>,
  <div
    className="button-panel main"
  >
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="button-panel main"
  >
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="button-panel main"
  >
    <button
      className="button main w-50"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "darkgray",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button main w-25"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="button"
    >
      =
    </button>
  </div>,
]
`;
